### PR TITLE
Add support for timestamp in headers

### PR DESCRIPTION
**Pull request by GwendalRaoul**
It uses the **kwargs argument we have for request/response/event If time_ms_epoch is None => no timestamp in protobuf object If time_ms_epoch is 0 => Automatically added in object creation (default) If time_ms_epoch is set => Used as is

### CODE REVIEW

File: tests/test_get_configs.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import wirepas_mesh_messaging
 from default_value import *
+import time
```
Comment (pgaucherand) [inline tests/test_get_configs.py:5]: this import is not used

File: wirepas_mesh_messaging/event.py [hunk 2]
```diff
@@ -18,17 +18,21 @@ class Event(object):
...
 
         super(Event, self).__init__()
         self.gw_id = gw_id
         self.sink_id = sink_id
         if event_id is None:
             event_id = random.getrandbits(64)
         self.event_id = event_id
+        if time_ms_epoch == 0:
+            time_ms_epoch = int(time.time() * 1000)
+        self.time_ms_epoch = time_ms_epoch
```
Comment (pgaucherand) [inline wirepas_mesh_messaging/event.py:35]: you can do this as a oneliner: 
self.time_ms_epoch = time_ms_epoch if time_ms_epoch!=0 else int(time.time() * 1000)

File: wirepas_mesh_messaging/received_data.py [hunk 3]
```diff
@@ -50,10 +51,11 @@ def __init__(
         event_id=None,
         hop_count=0,
         network_address=None,
+        time_ms_epoch=None,
         **kwargs
     ):
         super(ReceivedDataEvent, self).__init__(
-            gw_id, sink_id, event_id=event_id, **kwargs
+            gw_id, sink_id, event_id=event_id, time_ms_epoch=time_ms_epoch, **kwargs
```
Comment (pgaucherand) [inline wirepas_mesh_messaging/received_data.py:58]: do we need to overload the constructor with time_ms_epoch? We should be able to pass in the time_ms_epoch parameter without having to add this parameter in the super constructor call.

File: wirepas_mesh_messaging/request.py [hunk 4]
```diff
@@ -18,15 +17,19 @@ class Request(object):
...
 
     # pylint: disable=unused-argument
-    def __init__(self, sink_id=None, req_id=None, **kwargs):
+    def __init__(self, sink_id=None, req_id=None, time_ms_epoch=0, **kwargs):
         super(Request, self).__init__()
         self.sink_id = sink_id
         if req_id is None:
             req_id = random.getrandbits(64)
         self.req_id = req_id
+        if time_ms_epoch == 0:
```
Comment (pgaucherand) [inline wirepas_mesh_messaging/request.py:30]: same comment as before for the oneliner

File: wirepas_mesh_messaging/response.py [hunk 5]
```diff
@@ -20,15 +20,19 @@ class Response(object):
...
 
     # pylint: disable=unused-argument
-    def __init__(self, req_id, gw_id, res, sink_id=None, **kwargs):
+    def __init__(self, req_id, gw_id, res, sink_id=None, time_ms_epoch=0, **kwargs):
         super(Response, self).__init__()
         self.gw_id = gw_id
         self.sink_id = sink_id
         self.req_id = req_id
         self.res = res
+        if time_ms_epoch == 0:
```
Comment (pgaucherand) [inline wirepas_mesh_messaging/response.py:33]: same comment as before for the oneliner

File: wirepas_mesh_messaging/set_config.py [hunk 6]
```diff
@@ -46,10 +46,12 @@ class SetConfigRequest(Request):
...
                       only relevant keys for new config has to be defined
         req_id (int): unique request id
+        time_ms_epoch(int): timestamp in ms of request generation
+
     """
 
-    def __init__(self, sink_id, new_config, req_id=None, **kwargs):
-        super(SetConfigRequest, self).__init__(req_id=req_id, **kwargs)
+    def __init__(self, sink_id, new_config, req_id=None, time_ms_epoch=None, **kwargs):
+        super(SetConfigRequest, self).__init__(req_id=req_id, time_ms_epoch=time_ms_epoch, **kwargs)
```
Comment (pgaucherand) [inline wirepas_mesh_messaging/set_config.py:54]: do we need to overload the constructor with time_ms_epoch? We should be able to pass in the time_ms_epoch parameter without having to add this parameter in the super constructor call.

File: wirepas_mesh_messaging/set_scratchpad_target.py [hunk 7]
```diff
@@ -35,13 +35,16 @@ class SetScratchpadTargetAndActionRequest(Request):
...
 
     def __init__(self,
                  sink_id,
                  target,
-                 req_id=None, **kwargs):
-        super(SetScratchpadTargetAndActionRequest, self).__init__(sink_id, req_id, **kwargs)
+                 req_id=None,
+                 time_ms_epoch=None,
+                 **kwargs):
+        super(SetScratchpadTargetAndActionRequest, self).__init__(sink_id, req_id, time_ms_epoch=time_ms_epoch, **kwargs)
```
Comment (pgaucherand) [inline wirepas_mesh_messaging/set_scratchpad_target.py:47]: do we need to overload the constructor with time_ms_epoch? We should be able to pass in the time_ms_epoch parameter without having to add this parameter in the super constructor call.

**Review by pgaucherand — COMMENTED** [review]
Couple of comments from me to be discussed

File: wirepas_mesh_messaging/event.py (hunk 2 above)
Comment (GwendalRaoul) [reply wirepas_mesh_messaging/event.py:35]: Better to keep as above.
And to be honest, I prefer the multi-line approach :-)

**Review by pgaucherand — APPROVED** [review]

**Pull request merged**